### PR TITLE
FFI: add functions to access key/uid signature(s).

### PR DESCRIPTION
**Pull request by ni4**
This PR adds some basic signature access functions which I needed for CLI.
Also it fixes `rnp_locate_key()` function to set result to NULL if key was not found (basically, it should be in a separate PR, needed this in `rnp_signature_get_signer()`).

### CODE REVIEW

File: src/lib/rnp.cpp [hunk 1]
```diff
@@ -4212,6 +4220,207 @@ rnp_key_get_uid_handle_at(rnp_key_handle_t key, size_t idx, rnp_uid_handle_t *ui
...
+    }
+    uint8_t keyid[PGP_KEY_ID_SIZE] = {0};
+    if (!signature_get_keyid(&handle->sig->sig, keyid)) {
+        *result = NULL;
+        return RNP_SUCCESS;
+    }
+
+    size_t hex_len = PGP_KEY_ID_SIZE * 2 + 1;
+    *result = (char *) malloc(hex_len);
+    if (*result == NULL) {
```
Comment (dewyatt) [inline src/lib/rnp.cpp]: Style-wise I would argue to use `if (!*result)` though I'm not sure we have an official guideline on it.

Comment (ronaldtse) [reply src/lib/rnp.cpp]: I don’t think we have this specified in the guide, but I agree that your suggestion is more consistent (just a few lines above).

**Comment by ronaldtse** [comment]
@ni4 please feel free to merge after perhaps a tweak of the line mentioned above. Thanks!

File: src/lib/rnp.cpp (hunk 1 above)
Comment (ni4) [reply src/lib/rnp.cpp]: Thanks, you are right. Will fix and merge this soon.

**Comment by ronaldtse** [comment]
Thanks @ni4 !

**Pull request merged**